src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -86,48 +82,6 @@ def exception():
   def _map_exit_code(cls, value):
     return 0 if value in cls._SUCCESS_EXIT_CODES else value
 
-  @classmethod
-  def rc(cls, value):
-    exit_code = cls._map_exit_code(value)
-    return PytestResult('SUCCESS' if exit_code == 0 else 'FAILURE', rc=exit_code)
-
-  @classmethod
-  def from_error(cls, error):
-    if not isinstance(error, TaskError):
-      raise AssertionError('Can only synthesize a {} from a TaskError, given a {}'
-                           .format(cls.__name__, type(error).__name__))
-    return cls.rc(error.exit_code).with_failed_targets(error.failed_targets)
-
-  def with_failed_targets(self, failed_targets):
-    return PytestResult(self._msg, self._rc, failed_targets)
-
-  def __init__(self, msg, rc=None, failed_targets=None):
-    self._rc = rc
-    self._msg = msg
-    self._failed_targets = failed_targets or []
-
-  def __str__(self):
-    return self._msg
-
-  @property
-  def success(self):
-    return self._rc == 0
-
-  @property
-  def failed_targets(self):
-    return self._failed_targets
-
-  def checked(self):
-    """Raise if this result was unsuccessful and otherwise return this result unchanged.
-
-    :returns: this instance if successful
-    :rtype: :class:`PytestResult`
-    :raises: :class:`ErrorWhileTesting` if this result represents a failure
-    """
-    if not self.success:
-      raise ErrorWhileTesting(exit_code=self._rc, failed_targets=self._failed_targets)
-    return self
-
 
 class PytestRun(TestRunnerTaskMixin, Task):
```

Comment:
Completely out of left field, but: is there any more partitioning-related functionality that can be lifted into `TestRunnerTaskMixin`?

Reply:
There certainly is, PytestRun and JUnitRun do much of the same work for partitioning / caching / invalidation. I've filed #5307